automation_tools/repository.py

```diff
@@ -0,0 +1,156 @@
+            warn_only=True)
+
+
+def enable_repos(*args, **kwargs):
+    """Enable repos passed as ``args`` using ``subscription-manager repos
+    --enable``.
+
+    For example::
+
+        enable_repos('repo1', 'repo2')
+
+    Will run the command ``subscription-manager repos --enable repo1 repo2``.
+
+    """
+    run('subscription-manager repos --enable {0}'
+        .format(' '.join(['"{0}"'.format(repo) for repo in args])))
+
+
+def create_custom_repos(**kwargs):
+    """Create custom repofiles.
+
+    Each ``kwargs`` item will result in one repository file created. Where the
+    key is the repository filename and repository name, and the value is the
+    repository URL.
+
+    For example::
+
+        create_custom_repo(custom_repo='http://repourl.domain.com/path')
+
+    Will create a repository file named ``custom_repo.repo`` with the following
+    contents::
+
+        [custom_repo]
+        name=custom_repo
+        baseurl=http://repourl.domain.com/path
+        enabled=1
+        gpgcheck=0
+
+    """
+    for name, url in kwargs.items():
+        repo_file = StringIO()
+        repo_file.write(
+            '[{name}]\n'
+            'name={name}\n'
+            'baseurl={url}\n'
+            'enabled=1\n'
+            'gpgcheck=0\n'
+            .format(name=name, url=url)
+        )
+        put(local_path=repo_file,
+            remote_path='/etc/yum.repos.d/{0}.repo'.format(name))
+        repo_file.close()
+
+
+def enable_satellite_repos(cdn=False, disable_enabled=True):
+    """Enable repositories required to install Satellite 6
+
+    If ``disable_enabled`` is ``True``, then this task will first disable any
+    already enabled repository, including beaker ones.
+
+    """
+    if isinstance(cdn, str):
+        cdn = (cdn.lower() == 'true')
+    if isinstance(disable_enabled, str):
+        disable_enabled = (disable_enabled.lower() == 'true')
+
+    if disable_enabled is True:
+        disable_beaker_repos()
+        disable_repos('*')
+
+    # Disable yum plugin for sub-man
+    run('sed -i -e "s/^enabled.*/enabled=0/" '
+        '/etc/yum/pluginconf.d/subscription-manager.conf')
+
+    repos = [
+        'rhel-{0}-server-rpms',
+        'rhel-server-rhscl-{0}-rpms',
+    ]
+
+    if cdn is True:
+        repos.append('rhel-{0}-server-satellite-6.0-rpms')
+
+    enable_repos(*[repo.format(distro_info()[1]) for repo in repos])
+
+    run('yum repolist')
+
+
+def disable_beaker_repos():
+    """Disable beaker repositories
+
+    If yum-config-manager is available this task will disable the repos, if not
+    it will move the beaker repo files to the running user home directory
+
+    """
+    # Clean up system if Beaker-based
+    result = run('which yum-config-manager', warn_only=True)
+    if result.succeeded:
+        run('yum-config-manager --disable "beaker*"')
+    else:
+        run('mv /etc/yum.repos.d/beaker-* ~/', warn_only=True)
+    run('rm -rf /var/cache/yum*')
+
+
+def manage_custom_repos(**kwargs):
```

Comment:
Your assumption is correct. Custom repositories cannot be enabled/disabled with ``subscription-manager``